### PR TITLE
Fix BearerTokenLoggingFeature annotation detection and loading

### PR DESCRIPTION
**Pull request by robert3005**
# Before this PR
You couldn't use BearerTokenLoggingFeature due to class loading cycles and annotation detection not working for interfaces
# After this PR
Both should work correctly

### CODE REVIEW

**Comment by robert3005** [comment]
@iamdanfox

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java [hunk 1]
```diff
@@ -37,10 +37,6 @@
 public class BearerTokenLoggingFilter implements ContainerRequestFilter {
     private static final Logger log = LoggerFactory.getLogger(BearerTokenLoggingFilter.class);
 
-    public static final String USER_ID_KEY = Utilities.Key.USER_ID.getMdcKey();
-    public static final String SESSION_ID_KEY = Utilities.Key.SESSION_ID.getMdcKey();
-    public static final String TOKEN_ID_KEY = Utilities.Key.TOKEN_ID.getMdcKey();
```
Comment (iamdanfox) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: Given that these were public static final, could we just leave them as they used to be and annotate them @deprecated? We don't want to accidentally break someone!

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java [hunk 2]
```diff
@@ -54,8 +50,4 @@ public final void filter(ContainerRequestContext requestContext) {
         Optional<UnverifiedJsonWebToken> parsedJwt = UnverifiedJsonWebToken.tryParse(rawAuthHeader);
         Utilities.recordUnverifiedJwt(requestContext, parsedJwt);
     }
-
-    public static String getRequestPropertyKey(String key) {
```
Comment (iamdanfox) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: Same here- would prefer not to delete if possible

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/Utilities.java [hunk 3]
```diff
@@ -52,6 +52,10 @@ private static void setUnverifiedContext(ContainerRequestContext requestContext,
         }
     }
 
+    public static String getRequestPropertyKey(String key) {
```
Comment (iamdanfox) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/Utilities.java]: package private?

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java (hunk 1 above)
Comment (iamdanfox) [reply auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: A code search internally found people relying on this:

```
Object userId = requestContext.getProperty(
BearerTokenLoggingFilter.getRequestPropertyKey(BearerTokenLoggingFilter.USER_ID_KEY));
```

Comment (robert3005) [reply auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: ok, we can keep just have to duplicate

**Review by iamdanfox — APPROVED** [review]

**Pull request merged**